samtranslator/model/apigatewayv2.py

```diff
@@ -34,7 +34,7 @@ def assign_tags(self, tags: Dict[str, Any]) -> None:
         :param tags: Tags to be assigned to the resource
 
         """
-        if tags is not None and "Tags" in self.property_types:
+        if tags is not None and "Tags" in self.property_types and "Body" not in self.property_types:
```

Comment:
`Body` is always in the property_types, so this is essentially do nothing.

Reply:
I also notice my code `"Tags" in self.property_types` is a useless check too.

Reply:
Hmm that is a good point, I kind of misunderstood/misread while debugging. Just dug around the code a bit more and it looks like `Tags` can only be defined when `DefinitionBody` is specified or else we raise an [error](https://github.com/gracelu0/serverless-application-model/blob/develop/samtranslator/model/api/http_api_generator.py#L499). Also, `DefinitionBody` will always be defined because we have a [plugin](https://github.com/gracelu0/serverless-application-model/blob/develop/samtranslator/plugins/api/default_definition_body_plugin.py#L37) that generates a skeleton OpenApi definition before the transform. Additionally `Tags` always gets [added](https://github.com/gracelu0/serverless-application-model/blob/develop/samtranslator/model/api/http_api_generator.py#L522) to the `DefinitionBody`. 

So it seems like there is no case where we would need to add `Tags` at all which means `assign_tags()` in ApiGatewayV2 should just do nothing. Does that make sense?

Reply:
I think that makes sense.